tests/validation/tests/v3_api/test_pipeline.py

```diff
@@ -0,0 +1,100 @@
+from .common import get_global_admin_client_and_cluster, create_kubeconfig, \
+    random_test_name, create_project_and_ns, get_project_client_for_token, \
+    random_str, wait_for_condition, get_cluster_client_for_token, \
+    create_connection, WebsocketLogParse, CATTLE_TEST_URL, USER_TOKEN
+import pytest
+import urllib
+from threading import Lock
+import time
+
+#RANCHER_CLUSTER NAME, CATTLE_TEST_URL, USER_TOKEN
+namespace = {"p_client": None, "ns": None, "cluster": None,
+             "project": None, "pipeline": None, "pipeline_run": None}
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+    client, cluster = get_global_admin_client_and_cluster()
+    create_kubeconfig(cluster)
+    p, ns = create_project_and_ns(
+        USER_TOKEN, cluster, random_test_name("testworkload"))
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+
+    def fin():
+        #project delete does not remove all ns created in test, removing manually
+        client = get_cluster_client_for_token(namespace["cluster"], USER_TOKEN)
+        for ns_x in client.list_namespace(projectId=namespace["project"].id): 
+            client.delete(ns_x)
+        client.delete(namespace["project"])
+        
+    request.addfinalizer(fin)
+
+
+def check_last_run_state(status):
+    def _find_condition(resource):
+        if not hasattr(resource, "lastRunState"):
+            return False
+
+        if resource.lastRunState is None:
+            return False
+
+        if resource.lastRunState == status:
+            return True
+        if resource.lastRunState == "Failed": 
+            return False
+        return False
+    return _find_condition
+
+def create_example_pipeline():
+    return namespace["p_client"].create_pipeline(
+        name="test-" + random_str(),
+        repositoryUrl="https://github.com/rancher/pipeline-example-go.git",
+        triggerWebhookPr=False,
+        triggerWebhookPush=False,
+        triggerWebhookTag=False)
+
+def test_run_example_pipeline():
+    pipeline = create_example_pipeline()
+    assert len(namespace["p_client"].list_pipeline(projectId=namespace["project"].id).data) > 0
+    print("Created Pipeline, running example...")
+    namespace["pipeline"] = pipeline
+    namespace["pipeline_run"] = namespace["p_client"].action(obj=pipeline,
+                                action_name='run',
+                                branch="master")
+    wait_for_condition(namespace["p_client"], pipeline, check_last_run_state("Success"), timeout=600)
```

Comment:
Can you run a flake8 against the file?

Reply:
Can we have the `timeout` defined as a global variable here? Or can we use anything existing from common.py ?

Reply:
yes, added global var